File: Siddhi-Kawade/Armstrong.java
import java.util.*;
public class Armstrong {
  public static void main(String args[]){
    int num = 153;
    if(isArmstrong(num)){
      System.out.println(num + "Armstrong " );
    }
    else{
      System.out.println(num + " Not Armstrong ");
    } 
  }


public static boolean isArmstrong(int n){
  int sum=0;
  int duplicate=n;
  while(n>0){
    int digit=n%10;
    sum=sum+(digit*digit*digit);
    n=n/10;
  }
  return sum==duplicate;
}
}

File: Siddhi-Bodake/day3/armstrong.py
def isArmstrong(n):
    # Step 1: Find the number of digits
    num_digits = len(str(n))
    
    # Step 2: Initialize sum
    total = 0
    temp = n  # make a copy so we can modify it
    
    # Step 3: Extract digits and calculate powered sum
    while temp > 0:
        digit = temp % 10          # get last digit
        total += digit ** num_digits  # add its power
        temp //= 10                # remove last digit
    
    # Step 4: Check if sum equals original number
    return total == n


# Driver code
N = 153
if isArmstrong(N):
    print(N, "is an Armstrong number ✅")
else:
    print(N, "is NOT an Armstrong number ❌")
